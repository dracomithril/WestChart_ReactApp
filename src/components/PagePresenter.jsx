/**
 * Created by Gryzli on 08.04.2017.
 */
import React from 'react';
import PickYourDate from "./PickYourDate";
import MainMenu from "./Menu";
import ChartPresenter from "./ChartPresenter";
import {Modal} from "react-bootstrap";
import FilteringOptions from "./FilteringOptions";
import SongsPerDay from "./SongsPerDay";
let utils = require('./../utils');

const WaitMessage = (props) => {
    return (<Modal show={props.show}>
        <Modal.Header>{"Hello"}</Modal.Header>
        <Modal.Body>{`Hi, ${props.user_name} we are fetching data please wait.`}</Modal.Body>
    </Modal>)
};
export default class PagePresenter extends React.Component {
    componentWillUnmount() {
        console.log('component PagePresenter unmounted');
    }

    componentDidMount() {
        console.log('component PagePresenter did mount');
    }
    render() {
        const {store} = this.context;
        const {user, chart} = store.getState();
        let view_chart1 = [], error_days1 = [];
        if (chart.length > 0) {
            let {view_chart, error_days} = utils.filterChart(store);
            view_chart1 = view_chart;
            error_days1 = error_days;
        }
        return (<div>
            <WaitMessage show={chart.length === 0} user_name={user.first_name}/>
            {user.isGroupAdmin && <MainMenu/>}
            {user.isGroupAdmin &&
            <div>
                <div className="formArea">
                    <SongsPerDay error_days={error_days1}/>
                    <PickYourDate />
                    <FilteringOptions />
                </div>
                <ChartPresenter view_chart={view_chart1}/>
            </div>}

        </div>);
    }
}
PagePresenter.contextTypes={
    store:React.PropTypes.object
};
PagePresenter.propTypes = {};