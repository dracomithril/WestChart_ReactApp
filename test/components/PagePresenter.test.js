/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
jest.mock('./../../src/utils');
import configureMockStore from "redux-mock-store";
const initial_state = require('../data/initial_state.json');
const chartData = require('../data/response.json').chart;
import PagePresenter from "../../src/components/PagePresenter";

const mockStore = configureMockStore([]);

describe('<PagePresenter/>', () => {
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <PagePresenter />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.children().node).toBeUndefined();
    });
    it('renders with isGroupAdmin=true', () => {
        let state = Object.assign({},initial_state,{user:{isGroupAdmin:true}});
        const store = mockStore(state);
        const wrapper = shallow(
            <PagePresenter />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.children().props().id).toBe("groupAdmin");
    });
    it('should obtain chart objects', () => {
        const utils = require('./../../src/utils');
        utils.filterChart.mockReturnValue([]);
        let state = Object.assign({},initial_state,{user:{isGroupAdmin:true}, chart:chartData});
        const store = mockStore(state);
        const wrapper = shallow(
            <PagePresenter />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.children().props().id).toBe("groupAdmin");
    });
});
