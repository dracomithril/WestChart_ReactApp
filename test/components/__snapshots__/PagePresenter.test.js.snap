// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagePresenter/> renders with isGroupAdmin=true 1`] = `
<div>
  <div
    id="groupAdmin"
  >
    <div
      className="formArea"
    >
      <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={false}
      >
        <ModalHeader
          bsClass="modal-header"
          closeButton={false}
          closeLabel="Close"
        >
          Information
        </ModalHeader>
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
        >
          undefined we are fetching data please wait.
        </ModalBody>
      </Modal>
      <SongsPerDay
        error_days={Array []}
      />
      <PickYourDate />
      <FilteringOptions />
    </div>
    <ChartPresenter
      view_chart={Array []}
    />
  </div>
</div>
`;

exports[`<PagePresenter/> renders without crashing 1`] = `<div />`;

exports[`<PagePresenter/> should obtain chart objects 1`] = `
<div>
  <div
    id="groupAdmin"
  >
    <div
      className="formArea"
    >
      <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={false}
      >
        <ModalHeader
          bsClass="modal-header"
          closeButton={false}
          closeLabel="Close"
        >
          Information
        </ModalHeader>
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
        >
          undefined we are fetching data please wait.
        </ModalBody>
      </Modal>
      <SongsPerDay
        error_days={
          Array [
            Object {
              "color": "blue",
              "count": 1,
              "org": "2017-03-07T23:00:00.000Z",
            },
          ]
        }
      />
      <PickYourDate />
      <FilteringOptions />
    </div>
    <ChartPresenter
      view_chart={
        Array [
          Object {
            "added_by": "Joanna Kozieńska",
            "added_time": "2017-03-07T23:00:00.000Z",
            "created_time": "2017-03-07T23:07:03+0000",
            "from_user": "Joanna Kozieńska",
            "full_picture": "https://external.xx.fbcdn.net/safe_image.php?d=AQCimze9kDlMKVl1&w=720&h=720&url=https%3A%2F%2Fi.ytimg.com%2Fvi%2FLuxpfnnsdX4%2Fmaxresdefault.jpg&cfs=1&_nc_hash=AQC7M9AEfyBbtF3g",
            "id": 5,
            "likes_num": 8,
            "link": Object {
              "name": "Chet Faker - 1998 ft Banks",
              "title": "Chet Faker - 1998 ft Banks",
              "type": "video_share_youtube",
              "url": "https://youtu.be/LuxpfnnsdX4",
            },
            "message": "Jedna ode mnie, bo już wieki nie miałam okazji ;) 
        
        ",
            "reactions_num": 9,
            "selected": false,
            "source": "https://www.youtube.com/embed/LuxpfnnsdX4?autoplay=1",
            "type": "video",
            "updated_time": "2017-03-08T21:36:01+0000",
          },
        ]
      }
    />
  </div>
</div>
`;
